docs/tutorials/byproduct_recovery_determination.ipynb

```diff
@@ -0,0 +1,706 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "ba3193ce-0e23-44d0-a28a-959efe0acb48",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Pyomo packages\n",
+    "from pyomo.environ import (\n",
+    "    ConcreteModel,\n",
+    "    Expression,\n",
+    "    Param,\n",
+    "    TransformationFactory,\n",
+    "    Var,\n",
+    "    value,\n",
+    ")\n",
+    "\n",
+    "# IDAES packages\n",
+    "from idaes.core.util.model_diagnostics import DiagnosticsToolbox\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "\n",
+    "import pytest\n",
+    "\n",
+    "# PrOMMiS packages\n",
+    "from prommis.nanofiltration.diafiltration import (\n",
+    "    main,\n",
+    "    add_costing,\n",
+    "    add_objective,\n",
+    "    add_product_constraints,\n",
+    "    build_model,\n",
+    "    initialize_model,\n",
+    "    print_information,\n",
+    "    set_scaling,\n",
+    "    solve_model,\n",
+    "    unfix_opt_variables,\n",
+    ")\n",
+    "from prommis.uky.costing.costing_dictionaries import load_default_sale_prices\n",
+    "from prommis.uky.costing.determine_byproduct_recovery import (\n",
+    "    ByproductRecovery,\n",
+    "    determine_example_usage,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "04c9f0b2-437d-4791-a4a0-950471099cbf",
+   "metadata": {},
+   "source": [
+    "## __Step 2: Flowsheet Puilding__\n",
+    "\n",
+    "The lithium and colbolt diafiltration process has already been built and available in the repository: https://github.com/prommis/prommis/blob/main/src/prommis/nanofiltration/diafiltration.py.\n",
+    "\n",
+    "To determine whether lithium and colbolt should be recovered, you'll first need to __import the diafiltration process__ and __access the cost and revenue__.\n",
+    "\n",
+    "You can do this by referencing the process model defined in ``diafiltration.py``. Specifically, call the functions in the order specified within the ``main()`` function. Following this order is crucial because the diafiltration process consists of __three membrane stages__, and the model is built stage by stage. \n",
+    "\n",
+    "__Key Notes__:\n",
+    "- During each stage of model building:\n",
+    "\n",
+    "    - Initialization is performed with a set of variables fixed.\n",
+    "    - When transitioning to the next stage:\n",
+    "        - Previously fixed variables must be unfixed.\n",
+    "        - New variables need to be fixed for the current stage.\n",
+    "\n",
+    "This sequential initialization ensures that the model is stable and properly configured at each membrane stage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "65a60bef-f979-4021-91e6-a1bc6a146e6a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "WARNING (W1002): Setting Var 'fs.stage1.permeate[0.0,1].flow_vol' to a numeric\n",
```

Comment:
Are these warnings thrown by the model? @mollydougher do you have a suggestion to prevent these?